### ISSUE TITLE
Watch it! Movie watch list Submission

### ISSUE THREAD

**Issue by Vl-Wouter** [CodingGarden/seedling-school-01-frontend-project#2]
## Project Name / Title

Watch it! 

## Your Name / Title

Wouter

## Project Description

An application where users can look up movies/series and add them to their watch list. A user can also mark if they have seen the movie or not on their list.

*  https://github.com/Vl-Wouter/watch-it-movie-watch-list/ (description file found in the docs folder)

## What 3rd Party Web API do you plan to use?

* http://www.omdbapi.com/

## Which of the following describes you:

* [x] YouTube Subscriber
* [x] Twitch Follower
* [ ] Patreon Patron
* [ ] Superchat Donor
* [ ] Streamlabs Donor
* [ ] Coding Garden Moderator

**Comment by Vl-Wouter** [comment]
Submitted / Description under review. 

Committed the description so that can be checked before I start working on it.

**Comment by w3cj** [comment]
Looks good! Since you plan to allow users to save favorites or mark as watched, you will need to use local storage, index db or websql. Please decide and add that to your tech stack section and then you can get started designing / coding!

**Comment by Vl-Wouter** [comment]
Thanks! Forgot to add that. I updated it now.

**Comment by w3cj** [comment]
How are things so far? Any updates to share?

**Comment by Vl-Wouter** [comment]
Yes, I have finished the first designs (can also be found in the docs folder) and I'm now working on finishing the code

**Comment by Vl-Wouter** [comment]
Redid the wireframes: 
![Home with list](https://raw.githubusercontent.com/Vl-Wouter/watch-it-movie-watch-list/master/docs/wireframes/Home%20%20w-%20List.png)
![Home with intro](https://raw.githubusercontent.com/Vl-Wouter/watch-it-movie-watch-list/master/docs/wireframes/Home.png)

Fixed up some code and deployed a first version to Github Pages [here](https://vl-wouter.github.io/watch-it-movie-watch-list/index.html)!

Would love to get a code review and some UI/UX feedback on it. 😄

**Comment by w3cj** [comment]
Looks good so far! Nice font choice and color choice.

**Comment by Vl-Wouter** [comment]
Hey CJ,

I made some more fixes and improvements to the app and feel like I'm happy enough for it to get a code review and UI/UX feedback. (Already asked that before, but noticed I had some uncommitted changes that I really wanted to get to before feedback).

**Comment by w3cj** [comment]
## UI Feedback

Overall the app looks great. Great color and font choices. Nice subtle animations.

### Home Page

#### Things I Like

* Great color scheme choice
* I like the full screen background image

#### Things to improve

* The header / info font size could be a little bigger
* The scroll to the info section is a bit abrupt. You can add the `scroll-behavior` property to your html to make it smooth:

```css
html {
  scroll-behavior: smooth;
}
```

### App page

#### Things I Like

* The hover effect on the image search results is nice
* The zoom out effect when clicking "seen it" is great

#### Things To Improve

* When my list is empty, it might look nice to have the search bar large and in the center of the page. It's not immediately obvious that the search bar is at the top.
* The movie posters should show the full poster instead of a square crop.
* It is not immediately obvious that clicking a movie will add it to the list. You might add a button to each movie with the text "add to my list" similar to how you have the "seen it" button on the my list page.
* Instead of immediately showing "my list" after adding a movie, it would be nice to keep the user on the search result page so they can continue adding movies to their list.
* It would be nice to see how many movies total are in my list while on the search results. You could add a number next to the "My List" button at the top with the number of movies in "my list".

### Other feedback

* The API also accepts a page parameter. For instance the search for star wars has a totalResults property of 459. You can get the next page of results by adding `&page=2` on to the url. It would be nice to see a list of pages you can go to at the bottom of the search results. Total pages would be `Math.ceil(totalResults / 10)` i.e. there are 10 results per page.

These are just notes. You don't have to implement them. Just things to think about.

I'll add a code review next, but I'm going to spend some time on UI reviews for other's projects before I get to code review yours.

**Comment by Vl-Wouter** [comment]
Thanks a lot for the feedback! I didn't even know scroll-behavior was a thing.

I'll work on implementing what I can 👍

**Comment by w3cj** [comment]
# Code Review Checklist

### README

* [x] Readme includes instructions for running the project locally
  * Might add / use webpack dev server to build / run the project all in one go locally.

### HTML

* [x] Consistent Indentation
* [x] Includes meta viewport tag for mobile devices  
  * `<meta name="viewport" content="width=device-width, initial-scale=1.0">`
* [x] CSS Links and Font Links are in the head of the document
* [x] JavaScript files are linked at the bottom of the page OR at the top with an onload function OR at the top with a defer attribute
* [x] Uses semantic tags where available
  * header, time, main, nav, article, section etc.
  * See this article for more: https://html.com/semantic-markup/
* [ ] Title / Header links back to home page
  * Header does not link anywhere
* [x] No extra elements. Only includes the necessary elements to make things work.

#### HTML NOTEs

* Line 15, 20 and 37 have the closing `>` on the next line. This is still valid HTML, but looks a little weird to me.
  * Alca says it's to remove whitespace. If so, COOL!

### JavaScript

* [x] Consistent semicolon usage. Either do or do not. There is no inbetween.
* [x] Consistent quote usage. Either `'` or `"`, don't mix.
* [x] Consistent indentation.
* [ ] Reasonable max line length. Wrap / reformat code when it gets too long.
  * There are a few places with lines longer than 100 chars. Might create some variables to bring the line length down.
* [x] Variable / Function names are clear and concise.
  * [x] No abbreviations.
* [x] Variable / Function naming convention is consistent. 
  * camelCase or PascalCase or snake_case
* [ ] Strings used more than once are in a variable.
  * There are quite a few LONG strings that could be put into variables.
* [x] Functions are as few lines as possible. Code reads like a sentence.
* [ ] Function names dictate intent.
  * `toggleSearchForm` in `interaction.js` could have a better name. This function is not exactly toggling the form, it is hitting API and showing the results on the page.
* [x] Nested loops avoided where possible.
* [x] Functions take in a reasonable number of parameters. Ideally 3 or less params. Otherwise, use an options object.
* [x] No extra variables.

#### Notes on Promises

* There is no need to create a new Promise if you are already using a function that returns a Promise. The `fetchFromInput` function can be re-written like so:

```js
const fetchFromInput = (data) => {
  const title = data.get('movieTitle');
  return fetch(`https://www.omdbapi.com/?apikey=c223f3f3&s=${title}`)
    .then((response) => response.json())
    .then((res) => {
      const { Search: search } = res;
      if (search && search.length > 0) {
        return filterResults(search);
      } else {
        return [];
      }
    });
};
```

Also, you can re-write this as an async function:

```js
const fetchFromInput = async (data) => {
  console.log('using async await now...');
  const title = data.get('movieTitle');
  const response = await fetch(`https://www.omdbapi.com/?apikey=c223f3f3&s=${title}`)
  const { Search: search } = await response.json();
  if (search && search.length > 0) {
    return filterResults(search);
  }
  return [];
};
```

### CSS
* [x] Consistent indentation
* [x] Consistent naming convention
  * [ ] Uses app specific prefix where necessary
* [x] No duplicate styles. Re-use where applicable.

#### Feedback

* Remove commented code

**Comment by Vl-Wouter** [comment]
And thanks for the code review! Only just saw the stream. Absolutely loved everyone suggesting movies to check the pluralisation 😄 (I used Need For Speed to test that btw). 

The HTML looking like that did indeed happen because of Prettier. Using that and ESLint helped me a lot with the consistency in this project. As for `toggleSearchForm`, that was an old function I rewrote and forgot to rename, will fix that for sure!

Will take all the feedback and make some fixes when I have some free time, since college is taking up quite a bit atm.
